src/canvas-extractor/tests/graphql/extractor/test_extractor_integration.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+# Licensed to the Ed-Fi Alliance under one or more agreements.
+# The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+# See the LICENSE and NOTICES files in the project root for more information.
+
+import os
+import pytest
+
+from pandas import DataFrame
+from pathlib import Path
+from sqlalchemy import create_engine
+
+from edfi_canvas_extractor.graphql.extractor import GraphQLExtractor
+from edfi_canvas_extractor.graphql.schema import query_builder
+
+from edfi_canvas_extractor.graphql.assignments import assignments_synced_as_df
+from edfi_canvas_extractor.graphql.courses import courses_synced_as_df
+from edfi_canvas_extractor.graphql.enrollments import enrollments_synced_as_df
+from edfi_canvas_extractor.graphql.sections import sections_synced_as_df
+from edfi_canvas_extractor.graphql.students import students_synced_as_df
+
+
+@pytest.fixture(scope="class")
+def gql_class():
+    CANVAS_BASE_URL = os.environ['CANVAS_BASE_URL']
+    CANVAS_ACCESS_TOKEN = os.environ['CANVAS_ACCESS_TOKEN']
+    START_DATE = "2021-01-01"
+    END_DATE = "2030-01-01"
+
+    gql = GraphQLExtractor(
+        CANVAS_BASE_URL,
+        CANVAS_ACCESS_TOKEN,
+        "1",
+        START_DATE,
+        END_DATE,
+        )
+    yield gql
+
+
+@pytest.fixture(scope="class")
+def graphql_class(gql_class):
+    query = query_builder(account_id=1)
+    data = gql_class.get_from_canvas(query)
+    gql_class.extract(data)
+
+    yield gql_class
+
+
+@pytest.fixture(scope="class")
+def test_db_fixture():
+    DB_FILE = "tests/graphql/test.db"
+    Path(DB_FILE).unlink(missing_ok=True)
+    yield create_engine(f"sqlite:///{DB_FILE}", echo=True)
+    # Path(DB_FILE).unlink(missing_ok=True)
+
+
+@pytest.mark.skip
+class TestExtractorIntegration:
+
+    def test_extractor_request(self, gql_class):
```

Comment:
SHOULD: We can refactor this to just assert the `has_data` flag instead of performing the query building and fetching from canvas operations again.

Reply:
The first test it's checking that is getting information before the run method, just to be clear that will remove the graphql_class and leave gql_class to add the run method and change test_extractor_request to assert if has data, correct?

Reply:
Yes. The idea is to only run the fetch request once and perform the assertions on the already fetched data. The below [comment](https://github.com/Ed-Fi-Alliance-OSS/LMS-Toolkit/pull/279#discussion_r1022900842) makes it clearer. In this case, I think the `has_data` flag is a good indicator if we actually get any data or not.